src/mmw/apps/core/decorators.py

```diff
@@ -0,0 +1,66 @@
+# -*- coding: utf-8 -*-
+from __future__ import print_function
+from __future__ import unicode_literals
+
+from django.utils.timezone import now
+
+from apps.core.models import (Job,
+                              RequestLog)
+
+
+def log_request(view):
+    """
+    Log the request and its response as a RequestLog model
+    """
+
+    def decorator(request, *args, **kwargs):
+        requested_at = now()
+
+        view_result = view(request, *args, **kwargs)
+
+        user = request.user if request.user.is_authenticated() else None
+
+        job_uuid = view_result.data.get('job', None)
+        job = None
+        if (job_uuid):
+            try:
+                job = Job.objects.get(uuid=job_uuid)
```

Comment:
This is possibly the most expensive call (outside the main `INSERT` below). Ultimately we'll need to associate `RequestLog`s with `Job`s, but this can be done in two places:

 * At the write stage, as done here, where the Job UUID is translated to the Job ID before being inserted
 * At the read stage, when a report is being prepared, translating the stored Job UUID to a Job ID

Since the writes are expected to be orders of magnitude more frequent than reads, we should defer converting the Job UUID to the Job ID until that read stage. We can join the two tables when generating a report.